pkg/csi_driver/node.go

```diff
@@ -266,6 +266,12 @@ func (s *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublish
 		return nil, status.Errorf(codes.Internal, "mkdir failed for path %q: %v", targetPath, err)
 	}
 
+	// map of flags stripped from mount options: if flag is 'true' do NOT pass mo to sidecar
+	// flag should only be set to 'false' if feature flag is enabled and is supported by given sidecar version
+	disallowedFlags := map[string]bool{
+		GCSFuseProfileFlag: !(s.driver.config.EnableGcsfuseProfilesInternal && isSidecarVersionSupportedForGivenFeature(gcsFuseSidecarImage, GCSFuseProfilesMinVersion)),
+	}
```

Comment:
This isn't very readable. Please put `!(s.driver.config.EnableGcsfuseProfilesInternal && isSidecarVersionSupportedForGivenFeature(gcsFuseSidecarImage, GCSFuseProfilesMinVersion))` on separate line with descriptive variable. 


```
shouldDisallowProfileFlag :=  !(s.driver.config.EnableGcsfuseProfilesInternal && isSidecarVersionSupportedForGivenFeature(gcsFuseSidecarImage, GCSFuseProfilesMinVersion))
disallowedFlags := map[string]bool{
		GCSFuseProfileFlag: shouldDisallowProfileFlag,
	}
```

nit: Also, I think we could move disallowedFlags to its own helper function that returns disallowedFlags map, as NodePublishVolume is already complicated enough, but I'll leave that up to you